src/pages/index.js

```diff
@@ -0,0 +1,302 @@
+    });
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
+api
+  .getUserInfo()
+  .then((user) => {
+    profileName.textContent = user.name;
+    profileDescription.textContent = user.about;
+    avatarImg.src = user.avatar;
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+const pencilIconImg = document.querySelector(".profile__pencil-icon");
+pencilIconImg.src = pencilIcon;
+const cardTemplate = document.querySelector("#card");
+const cardsList = document.querySelector(".cards__list");
+const logoImg = document.querySelector(".header__logo");
+logoImg.src = logo;
+
+const previewModal = document.querySelector("#preview-modal");
+const previewImage = previewModal.querySelector(".modal__image");
+const previewCaption = previewModal.querySelector(".modal__caption");
+const previewCloseBtn = previewModal.querySelector(".modal__close-btn");
+
+const editModal = document.querySelector("#edit-modal");
+const addCardModal = document.querySelector("#add-card-modal");
+const avatarModal = document.querySelector("#avatar-modal");
+const avatarForm = avatarModal.querySelector(".modal__form");
+const avatarModalCloseBtn = avatarModal.querySelector(".modal__close-btn");
+const avatarInput = avatarModal.querySelector("#profile-avatar-input");
+const deleteModal = document.querySelector("#delete-modal");
+const deleteForm = deleteModal.querySelector(".modal__form");
+const deleteModalCancelBtn = deleteModal.querySelector(
+  ".modal__button_type_cancel"
+);
+
+const profileName = document.querySelector(".profile__name");
+const profileDescription = document.querySelector(".profile__description");
+const profileEditBtn = document.querySelector(".profile__edit-button");
+const profileAddBtn = document.querySelector(".profile__add-button");
+const avatarModalBtn = document.querySelector(".profile__avatar-btn");
+const avatarImg = document.getElementById("profile-avatar");
+avatarImg.src = avatarSrc;
+
+const editForm = editModal.querySelector(".modal__form[name='edit-profile']");
+const addCardForm = addCardModal.querySelector(".modal__form[name='add-card']");
+
+const editNameInput = editForm.querySelector("#name");
+const editDescriptionInput = editForm.querySelector("#description");
+const addCardLinkInput = addCardForm.querySelector("#add-card-link");
+const addCardCaptionInput = addCardForm.querySelector("#add-card-caption");
+
+let selectedCard;
+let selectedCardId;
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", handleEscClose);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", handleEscClose);
+}
+function handleDeleteCard(cardElement, data) {
+  selectedCard = cardElement;
+  selectedCardId = data._id;
+  openModal(deleteModal);
+}
+
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+
+  const deleteBtn = deleteForm.querySelector(".modal__button_type_delete");
```

Comment:
It’s better to find **constant DOM elements** only 1 time at the top of the file so as not to waste resources on searching them again and again when you call a method (function) because searching is a very **hard** operation for the browser engine